src/libraries/System.Net.Security/tests/FunctionalTests/ServerAsyncAuthenticateTest.cs

```diff
@@ -69,6 +69,159 @@ public async Task ServerAsyncAuthenticate_EachSupportedProtocol_Success(SslProto
+            var clientOptions = new SslClientAuthenticationOptions() { TargetHost = _serverCertificate.GetNameInfo(X509NameType.SimpleName, false) };
+            clientOptions.RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
+
+            (SslStream client, SslStream server) = TestHelper.GetConnectedSslStreams();
+            using (client)
+            using (server)
+            {
+                Task t1 = client.AuthenticateAsClientAsync(clientOptions, default);
+                Task t2 = server.AuthenticateAsServerAsync(
+                    (stream, clientHelloInfo, userState, cancellationToken) =>
+                    {
+                        Assert.Equal(server, stream);
+                        Assert.Equal(clientOptions.TargetHost, clientHelloInfo.ServerName);
+                        return new ValueTask<SslServerAuthenticationOptions>(serverOptions);
+                    },
+                    null, default);
+
+                await TestConfiguration.WhenAllOrAnyFailedWithTimeout(t1, t2);
+                // Verify that the SNI callback can impact version.
+                Assert.Equal(version, client.SslProtocol);
+            }
+        }
+
+        private async Task<SslServerAuthenticationOptions> FailedTask()
+        {
+            await Task.Yield();
+            throw new InvalidOperationException("foo");
+        }
+
+        private async Task<SslServerAuthenticationOptions> OptionsTask(SslServerAuthenticationOptions value)
+        {
+            await Task.Yield();
+            return value;
+        }
+
+        [Fact]
+        public async Task ServerAsyncAuthenticate_AsyncOptions_Success()
+        {
+            var state = new Object();
+            var serverOptions = new SslServerAuthenticationOptions() { ServerCertificate = _serverCertificate };
+            var clientOptions = new SslClientAuthenticationOptions() { TargetHost = _serverCertificate.GetNameInfo(X509NameType.SimpleName, false) };
+            clientOptions.RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
+
+            (SslStream client, SslStream server) = TestHelper.GetConnectedSslStreams();
+            using (client)
+            using (server)
+            {
+                Task t1 = client.AuthenticateAsClientAsync(clientOptions, default);
+                Task t2 = server.AuthenticateAsServerAsync(
+                    (stream, clientHelloInfo, userState, cancellationToken) =>
+                    {
+                        Assert.Equal(server, stream);
+                        Assert.Equal(clientOptions.TargetHost, clientHelloInfo.ServerName);
+                        Assert.True(object.ReferenceEquals(state, userState));
+                        return new ValueTask<SslServerAuthenticationOptions>(OptionsTask(serverOptions));
+                    },
+                    state, default);
```

Comment:
```suggestion
                    state, CancellationToken.None);
```

(there's more of this elsewhere in the other tests, going to stop adding comments now!)